flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/NFA.java

```diff
@@ -1131,6 +1152,7 @@ private void serializeStates(Set<State<T>> states, DataOutputView out) throws IO
 					nameSerializer.serialize(transition.getTargetState().getName(), out);
 					actionSerializer.serialize(transition.getAction(), out);
 
+					// backward compatibility
 					serializeCondition(transition.getCondition(), out);
```

Comment:
Yes, but I think we could increase the VERSION field, and wrap this line in `if (readVersion == 1) {...}`